### PR TITLE
HIVE-28369: Fix LLAP cache proactive eviction

### PR DESCRIPTION
**Pull request by ngsg**
### What changes were proposed in this pull request?

This patch adds configuration/null checks to prevent NullPointerException.

### Why are the changes needed?

As I explained the JIRA ticket(HIVE-28369), Hive throws NPE during proactive eviction when `hive.llap.io.encode.enabled` is false. 

### Does this PR introduce _any_ user-facing change?

No

### Is the change a dependency upgrade?

No

### How was this patch tested?

Tested manually.
This cannot be tested using qtest because we cannot set up LLAP configuration in qfile as LLAP environment is initialized before test driver reads qfiles.

### CODE REVIEW

**Review by kokila-19 — APPROVED** [review]

File: llap-server/src/java/org/apache/hadoop/hive/llap/io/api/impl/LlapIoImpl.java [hunk 1]
```diff
@@ -308,9 +308,14 @@ public long evictEntity(LlapDaemonProtocolProtos.EvictEntityRequestProto protoRe
     LOG.debug("Starting proactive eviction.");
     long time = System.currentTimeMillis();
 
-    long markedBytes = dataCache.markBuffersForProactiveEviction(predicate, isInstantDeallocation);
-    markedBytes += fileMetadataCache.markBuffersForProactiveEviction(predicate, isInstantDeallocation);
-    markedBytes += serdeCache.markBuffersForProactiveEviction(predicate, isInstantDeallocation);
+    long markedBytes = 0;
+    if (useLowLevelCache) {
```
Comment (deniskuzZ) [inline llap-server/src/java/org/apache/hadoop/hive/llap/io/api/impl/LlapIoImpl.java:312]: `SimpleBufferManager` is used for `dataCache` when `useLowLevelCache` is `false`, so we should evict it as well

**Review by deniskuzZ — CHANGES_REQUESTED** [review]
`SimpleBufferManager` is used for `dataCache` when `useLowLevelCache` is `false`, so we should evict it as well

**Comment by ngsg** [comment]
> `SimpleBufferManager` is used for `dataCache` when `useLowLevelCache` is `false`, so we should evict it as well

@deniskuzZ,  `SimpleBufferManager#markBuffersForProactiveEviction` always throws `UnsupportedOperationException`, so I excluded it when `useLowLevelCache == false`.

In fact, we don't have to check `dataCache` and `fileMetadataCache` because the first if-clause (`memoryManager == null`) ensures that `dataCache instanceof LowLevelCacheImpl` and `fileMetadataCache != null`. I didn't notice it and fixed my patch to check only `serdeCache`.

**Review by deniskuzZ — APPROVED** [review]
LGTM +1

**Pull request merged**